RELEASE_CHECKLIST.md

```diff
@@ -0,0 +1,61 @@
+# Release Checklist
+
+## For every release
+
+- [ ] Regenerate LICENSE (see `license-generator/`)
+- [ ] Release notes
+- [ ] Publish to Hackage
+- [ ] Update npm package
+- [ ] Try PureScript? &mdash; need to decide whether we want to continue
+      officially supporting this
+
+## Libraries
+
+Are there breaking changes to the language? Or alternatively, are there
+language changes which require breaking changes in the relevant libraries to
+make use of? If so:
+
+- [ ] Update core libraries
+- [ ] Update contrib libraries
+- [ ] Update node bindings
+- [ ] Create a new package set
+
+## Tools
+
+Has the compiler CLI changed at all? If so, the following may need updates:
+
+- [ ] psc-package
+- [ ] Pulp
+- [ ] purs-loader
+- [ ] ide plugins
+
+## JSON formats
+
+Have any of the following JSON formats changed? If so, it may be worth
+considering what effects this may have:
+
+- [ ] Corefn
+- [ ] JSON produced by `purs publish`
+  - [ ] check whether this affects Pursuit
```

Comment:
The IDE protocol is another JSON format that might break (though the affected users are a smaller group as it's primarily plugin authors)